WW/ZZ detector and physics performance plots

beyerja opened the pull request:
Performance plots as in ILD LoI (6.6.3) showing separation power between hadronic WW and ZZ decays in vvqqqq samples added to ILDPerformance package.
Written such that it can be run for different detector models starting from the .stdhep files. The user only has to execute one script and modify one configuration file.



BEGINRELEASENOTES
- Adding WW/ZZ separation plots to ILDPerformance.

ENDRELEASENOTES

Commit:
WW/ZZ detector and physics performance plots

Performance plots as in ILD LoI (6.6.3) showing separation power between hadronic WW and ZZ decays in vvqqqq samples added to ILDPerformance package.
Written such that it can be run for different detector models starting from the .stdhep files. The user only has to execute one script and modify one configuration file.

Commit:
Debugged bash script, removed compiler warning

Debugged run_performance_check.sh so that it correctly calls scripts within scripts directory. Removed unnecessary memset from processor which caused compiler warnings.

rete commented on WWZZSeparation/processor/src/WWZZProcessor.cc:
```diff
@@ -0,0 +1,153 @@
+    _otfile->cd();
+	_tree->Write();
+    _otfile->Close();
+
```
Memory leak here. You should delete `_otfile` (but not the TTree which is owned by the TFile itself)

rete commented on WWZZSeparation/plotting/create_mjjmjjplots.cc:
```diff
@@ -0,0 +1,128 @@
+	string input_tree_name = "eventinfo"; 
+
+	// Histograms for WW/ZZ events
+	TH1* h1_WW = new TH1D("m_WW", 
```
Memory leak here. The histograms are not deleted at the end of the macro

rete commented on WWZZSeparation/plotting/create_mjjmjjplots.cc:
```diff
@@ -0,0 +1,128 @@
+
+
+	// Do drawing and saving of plots
+	TCanvas *canvas_h1 = new TCanvas("can_h1", "", 0, 0, 600, 600);
```
Memory leak on TCanvas

rete commented on WWZZSeparation/plotting/create_mjjmjjplots.cc:
```diff
@@ -0,0 +1,128 @@
+	string plot_name_h1 = input_directory + "/m.pdf";
+	canvas_h1->Print(plot_name_h1.c_str());
+
+	TCanvas *canvas_h2 = new TCanvas("can_h2", "", 0, 0, 800, 800);
```
Memory leak on TCanvas

Commit:
Avoiding memory leaks

Added delete statements on pointers to avoid memory leaks.

Commit:
Recommiting accidentally reset change

Had by accident overwritten an commit changing the run_performance_check.sh script. Recommitting it.
